File: backend/src/models/schemas/product.py
from decimal import Decimal

from backend.src.database import BaseSchema
from backend.src.models.enums.medium import Medium


class ProductBase(BaseSchema):
    id: int
    title: str
    price: Decimal
    height: int
    width: int
    medium_id: Medium
    thumbnail: str


class ProductsOut(ProductBase):
    image_url: str


class ProductOut(ProductBase):
    description: str | None
    images: list[str]

File: src/lib/interfaces/product.ts
interface Product {
    id: number;
    title: string;
    price: number;
    height: number;
    width: number;
    medium: "Painting" | "Print";
    thumbnail: string;
}

export interface ProductsOut extends Product {
    imageUrl: string;
}

export interface ProductOut extends Product {
    description: string | null;
    images: string[];
}